src/main/services/redlock.ts

```diff
@@ -0,0 +1,100 @@
+import * as Sentry from '@sentry/node'
+import Redlock from 'redlock'
+
+import { status } from '../../shared/status'
+import { createRedis } from '../../shared/utils'
+import { PluginsServer } from '../../types'
+
+type RedlockOptions = {
+    server: PluginsServer
+    resource: string
+    onLock: () => Promise<void> | void
+    onUnlock: () => Promise<void> | void
+    ttl: number
+}
+
+export async function startRedlock({
+    server,
+    resource,
+    onLock,
+    onUnlock,
+    ttl,
+}: RedlockOptions): Promise<() => Promise<void>> {
+    status.info('⏰', `Starting redlock "${resource}" ...`)
+
+    let stopped = false
+    let weHaveTheLock = false
+    let lock: Redlock.Lock
+    let lockTimeout: NodeJS.Timeout
+
+    const lockTTL = ttl * 1000 // 60 sec if default passed in
+    const retryDelay = lockTTL / 10 // 6 sec
+    const extendDelay = lockTTL / 2 // 30 sec
+
+    // use another redis connection for redlock
+    const redis = await createRedis(server)
+
+    const redlock = new Redlock([redis], {
+        // we handle retires ourselves to have a way to cancel the promises on quit
```

Comment:
retires or retries? 🤔

Reply:
🤔 indeed :) 

this was copypastad from the previous place, fixed now